Sources/Segment/Configuration.swift

```diff
@@ -45,60 +48,128 @@ public class Configuration {
 // MARK: - Analytics Configuration
 
 public extension Configuration {
+    
+    /// Sets a reference to your application.  This can be useful in instances
+    /// where referring back to your application is necessary, such as within plugins
+    /// or async code.  The default value is `nil`.
+    ///
+    /// - Parameter value: A reference to your application.
+    /// - Returns: The current Configuration.
     @discardableResult
     func application(_ value: Any?) -> Configuration {
         values.application = value
         return self
     }
     
+    /// Opt-in/out of tracking lifecycle events.  The default value is `false`.
+    ///
+    /// - Parameter enabled: A bool value
+    /// - Returns: The current Configuration.
     @discardableResult
     func trackApplicationLifecycleEvents(_ enabled: Bool) -> Configuration {
         values.trackApplicationLifecycleEvents = enabled
         return self
     }
     
+    /// Set the number of events necessary to automatically flush. The default
+    /// value is `20`.
+    ///
+    /// - Parameter count: Event count to trigger a flush.
+    /// - Returns: The current Configuration.
     @discardableResult
     func flushAt(_ count: Int) -> Configuration {
         values.flushAt = count
         return self
     }
     
+    /// Set a time interval (in seconds) by which to trigger an automatic flush.
+    /// The default value is `30`.
+    ///
+    /// - Parameter interval: A time interval
+    /// - Returns: The current Configuration.
     @discardableResult
     func flushInterval(_ interval: TimeInterval) -> Configuration {
         values.flushInterval = interval
         return self
     }
     
+    /// Sets a default set of Settings.  Normally these will come from Segment's
+    /// api.segment.com/v1/projects/<writekey>/settings, however in instances such
+    /// as first app launch, it can be useful to have a pre-set batch of settings to
+    /// ensure that the proper destinations and other settings are enabled prior
+    /// to receiving them from the Settings endpoint.  The default is `nil`.
+    ///
+    /// You can retrieve a copy of your settings from the following URL:
+    ///
+    /// https://cdn-settings.segment.com/v1/projects/<writekey>/settings
+    ///
+    /// Example:
+    /// ```
+    /// let defaults = Settings.load(resource: "mySegmentSettings.json")
+    /// let config = Configuration(writeKey: "1234").defaultSettings(defaults)
+    /// ```
+    ///
+    /// - Parameter settings: 
+    /// - Returns: The current Configuration.
     @discardableResult
-    func defaultSettings(_ settings: Settings) -> Configuration {
+    func defaultSettings(_ settings: Settings?) -> Configuration {
```

Comment:
I guess this means that values.defaultSettings can now be nil. Any downstream implications?

Reply:
Yup.  No downstream implications.  People weren't passing in nil before, so whatever they're passing now will continue.